compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -388,7 +386,7 @@ checkFetch tpl cid = do
 -- returns the contract id and contract type
 checkRetrieveByKey :: MonadGamma m => RetrieveByKey -> m (Type, Type)
 checkRetrieveByKey RetrieveByKey{..} = do
-  checkSupportsContractKeys
+  checkFeature featureContractKeys
```

Comment:
It looks like the changes here result in a call to `error` rather than a call to `throwWithContext`. It isn’t immediately obvious to me that this will be handled the same way and if not, what exactly this implies.

Reply:
Nope, this calls a different checkFeature which does call throwWithContext - so all good.